File: Softlithe.ERP.Api/Softlithe.ERP.Abstracciones/Contenedores/Marcas/MarcaDto.cs
using Softlithe.ERP.Abstracciones.Contenedores.MensajesDelSistema;
using Softlithe.ERP.Abstracciones.Contenedores.Monedas;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Softlithe.ERP.Abstracciones.Contenedores.Marcas
{
    public class MarcaDto
    {
        [Required(ErrorMessage = MensajeDeMarcaDto.CodigoMarcaRequerida)]
        public int NoMarca { get; set; }
        [Required(ErrorMessage = MensajesGeneralesDelSistemaDto.CodigoEmpresaRequerida)]
        public int NoEmpresa { get; set; }
        [Required(ErrorMessage = MensajeDeMarcaDto.NombreMarcaRequerida)]
        public string Descripcion { get; set; } = string.Empty;
        [Required(ErrorMessage = MensajesGeneralesDelSistemaDto.CodigoIdentificadorRequerido)]
        public int Identificador { get; set; }
        [Required(ErrorMessage = MensajesGeneralesDelSistemaDto.UsuarioRequerido)]
        public string Usuario { get; set; } = string.Empty;
        [Required(ErrorMessage = MensajesGeneralesDelSistemaDto.DatoEsActivoRequerido)]
        public bool EsActivo { get; set; }
    }
    public class MarcaConModeloDeValidacion : ModeloValidacion
    {
        public List<MarcaDto> LaListaDeMarcas { get; set; } = new List<MarcaDto>();
    }
}

File: SPA-Elixir/Desarrollo/src/requests/mantenimientos/marca/marcaDto.js
/**
 * Cuerpo JSON alineado a MarcaDto del API Softlithe (propiedades en PascalCase).
 */

export const construirMarcaDto = ({
    noMarca = 0,
    noEmpresa,
    identificador,
    descripcion,
    esActivo = true,
    usuario,
}) => ({
    NoMarca: Number.parseInt(String(noMarca), 10) || 0,
    NoEmpresa: Number.parseInt(String(noEmpresa), 10) || 0,
    Descripcion: descripcion != null ? String(descripcion) : '',
    Identificador: Number.parseInt(String(identificador), 10) || 0,
    EsActivo: !!esActivo,
    Usuario: usuario ?? '',
});
